packages/foundation/src/IComponent.ts

```diff
@@ -8,6 +8,11 @@ export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
 //        export const styles: IStackComponent['styles'] = props => {
 //        Existing issue: https://github.com/Microsoft/TypeScript/issues/241
 
+/**
+ * Helper interface for accessing user props children.
+ */
+export type IPropsWithChildren<TProps> = TProps & { children?: React.ReactNode };
```

Comment:
What's the rationale behind moving this here from `ISlots.ts`?

Reply:
To prevent creating a circular dependency between `ISlots.ts` and `IComponent.ts`.